interface/src/app/node/node.js

```diff
@@ -304,6 +307,32 @@ angular.module('adage.node', [
         $scope.numExpShown =
           $scope.topMode ? $scope.topNum : $scope.experiments.length;
       };
+
+      // Event handler when user clicks the checkbox in front of each sample.
+      // It keeps track of the number of selected samples in each experiment
+      // to indicate whether the "show annotations" button should be enabled.
+      $scope.toggleSelection = function(exp, selected) {
+        if (selected) {
+          ++exp.numSelections;
+        } else {
+          --exp.numSelections;
+        }
+      };
```

Comment:
Also, other than reference counting, we could possibly use the JavaScript array ``filter()`` function here, inside the ``$scope.toggleSelection()`` function. It could go something like:
```
var selectedSamples = exp.samples.filter(function(sample) {
  return (sample.selected === true);
}
if (selectedSamples.length > 0) {
  exp.selectedSamples = true;
} else {
  exp.selectedSamples = false;
}
```
or something along those lines. That is another possibility.